kafkakewl-metrics/src/main/scala/com/mwam/kafkakewl/metrics/domain/KafkaConsumerGroupMetrics.scala

```diff
@@ -0,0 +1,71 @@
+/*
+ * SPDX-FileCopyrightText: 2023 Marshall Wace <opensource@mwam.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.mwam.kafkakewl.metrics.domain
+
+enum ConsumerGroupStatus(val severity: Int):
+  /** Aggregates 2 ConsumerGroupStatuses so that the result is the more severe.
+    */
+  def +(other: ConsumerGroupStatus): ConsumerGroupStatus = if (severity < other.severity) other else this
+
+  case Ok extends ConsumerGroupStatus(1)
+  case Unknown extends ConsumerGroupStatus(2)
+  case Warning extends ConsumerGroupStatus(3)
+  case Error extends ConsumerGroupStatus(4)
+  case MaybeStopped extends ConsumerGroupStatus(5)
+  case Stopped extends ConsumerGroupStatus(6)
+
+end ConsumerGroupStatus
+
+implicit class ConsumerGroupStatusIterableExtensions(consumerGroupStatuses: Iterable[ConsumerGroupStatus]) {
+  def combine: ConsumerGroupStatus = consumerGroupStatuses.fold(ConsumerGroupStatus.Ok)(_ + _)
+}
```

Comment:
nitpick, but in scala 3 we should be using extensions instead of these implicit classes.

Reply:
Changed to be extensions